test/Job.test.js

```diff
@@ -23,97 +31,87 @@ const TEST_JOB_ID = 1;
 //  http://mochajs.org/#arrow-functions
 describe('Testing Job class', function() {
   before(function(done) {
-    server = GSF.server(config.localHTTPServer);
+    client = GSF.client(config.localHTTPServer);
     done();
   });
 
   afterEach(function(done) {
-    server.removeAllListeners('JobProgress');
+    client.removeAllListeners('JobProgress');
     done();
   });
 
   // ==========================================================
   describe('Job() constructor', function() {
     it('returns a valid job object', function(done) {
-      const job = new Job(server, TEST_JOB_ID);
+      const job = new Job(client, TEST_JOB_ID);
       expect(job).to.be.an('object');
       expect(job.jobId).to.equal(TEST_JOB_ID);
       done();
     });
   });
 
   describe('.jobInfo()', function() {
-    it('retrieves the job information', function(done) {
-      const task = new Task(server.service(testTasks.sleepTask.service),
+    it('retrieves the job information', function() {
+      const task = new Task(client.service(testTasks.sleepTask.service),
         testTasks.sleepTask.name);
-      task.submit({parameters: testTasks.sleepTask.parameters})
-        .then((job) => {
-          job.once('Completed', function(success) {
-            job.info()
-              .then((jobInfo) => {
-                expect(jobInfo).to.be.an('object');
-                expect(Array.isArray(jobInfo.results)).to.be.false;
-                verifyProperties(jobInfo, interfaces.jobInfo);
-                expect(jobInfo.jobStatus).to.equal('Succeeded');
-                expect(jobInfo.results).to.deep
-                  .equal(testTasks.sleepTask.results);
-                done();
-              }).catch((err) => {
-                done('Error getting job info: ' + err);
-              }).catch((err) => {
-                done('Error submitting job: ' + err);
-              });
-          });
+
+      let job = null;
+      return task
+        .submit({inputParameters: testTasks.sleepTask.parameters})
+        .then((jobObj) => {
+          job = jobObj;
+          return job.wait();
+        })
+        .then(() => {
+          return job
+            .info()
+            .then((jobInfo) => {
+              expect(jobInfo).to.be.an('object');
+              expect(jobInfo.jobResults).to.be.an.an('object');
+              verifyProperties(jobInfo, interfaces.jobInfo);
+              expect(jobInfo.jobStatus).to.equal('Succeeded');
+              expect(jobInfo.jobResults).to.deep
+                .equal(testTasks.sleepTask.results);
```

Comment:
no done call?

Reply:
Looks like there are several tests that removed the done callback. I'll just leave this one comment about it here.

Reply:
I assume the test framework supports promises? In anycase, as long as we are consistent I'm fine with it.

Reply:
That's correct, I removed them in favor of mocha's built in support for promises: https://mochajs.org/#asynchronous-code